lib/reverse_words.rb

```diff
@@ -1,4 +1,30 @@
 # A method to reverse each word in a sentence, in place.
 def reverse_words(my_words)
-  raise NotImplementedError
+  if my_words == nil
+    return nil
+  end
+  string_length = (my_words.length - 1)
+  count = 0
+  until count > string_length do
+    if my_words[count] =~ /\S/
+      word_start = count
+      word_end = count
+      until my_words[word_end] =~ /\s/ || word_end == string_length
+        word_end += 1
+      end
+      new_count = word_end
```

Comment:
Why not set count to the value of word_end? Do you need new_count as a separate variable?